libsql-server/src/namespace/meta_store.rs

```diff
@@ -0,0 +1,243 @@
+#![allow(dead_code)]
+
+use std::{collections::HashMap, path::Path, sync::Arc};
+
+use crate::{connection::config::DatabaseConfig, error::Error, Result};
+use parking_lot::Mutex;
+use rusqlite::Connection;
+use tokio::sync::{
+    mpsc,
+    watch::{self, Receiver, Sender},
+};
+
+use super::NamespaceName;
+
+type ChangeMsg = (NamespaceName, Arc<DatabaseConfig>);
+
+#[derive(Debug)]
+pub struct MetaStore {
+    changes_tx: mpsc::Sender<ChangeMsg>,
+    inner: Arc<Mutex<MetaStoreInner>>,
+}
+
+#[derive(Clone, Debug)]
+pub struct MetaStoreHandle {
+    namespace: NamespaceName,
+    inner: HandleState,
+}
+
+#[derive(Debug, Clone)]
+enum HandleState {
+    Internal(Arc<Mutex<Arc<DatabaseConfig>>>),
```

Comment:
Could we have Arc<Mutex<DatabaseConfig>>?

Reply:
Returned type expects an `Arc<DatabaseConfig>` so we unfortunately need to do this but that said this code should be test only and internal unit test so let me see if I can wrap in a cfg and we can forget about it until we refactor those tests.